flink-core/src/main/java/org/apache/flink/api/common/typeutils/base/LocalDateTimeComparator.java

```diff
@@ -0,0 +1,153 @@
+import org.apache.flink.core.memory.DataInputView;
+import org.apache.flink.core.memory.DataOutputView;
+import org.apache.flink.core.memory.MemorySegment;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.time.LocalDateTime;
+
+/**
+ * This class can not extend {@link BasicTypeComparator}, because LocalDateTime is a
+ * Comparable of ChronoLocalDateTime instead of Comparable of LocalDateTime.
+ */
+@Internal
+public final class LocalDateTimeComparator extends TypeComparator<LocalDateTime> implements Serializable {
+
+	private transient LocalDateTime reference;
+
+	protected final boolean ascendingComparison;
+	protected final LocalDateComparator dateComparator;
+	protected final LocalTimeComparator timeComparator;
+
+	// For use by getComparators
+	@SuppressWarnings("rawtypes")
+	private final LocalDateTimeComparator[] comparators = new LocalDateTimeComparator[] {this};
+
+	public LocalDateTimeComparator(boolean ascending) {
+		this.ascendingComparison = ascending;
+		this.dateComparator = new LocalDateComparator(ascending);
+		this.timeComparator = new LocalTimeComparator(ascending);
+	}
+
+	@Override
+	public int hash(LocalDateTime value) {
+		return value.hashCode();
+	}
+
+	@Override
+	public void setReference(LocalDateTime toCompare) {
+		this.reference = toCompare;
+	}
+
+	@Override
+	public boolean equalToReference(LocalDateTime candidate) {
+		return candidate.equals(reference);
+	}
+
+	@Override
+	public int compareToReference(TypeComparator<LocalDateTime> referencedComparator) {
+		int comp = ((LocalDateTimeComparator) referencedComparator).reference.compareTo(reference);
+		return ascendingComparison ? comp : -comp;
+	}
+
+	@Override
+	public int compare(LocalDateTime first, LocalDateTime second) {
+		int cmp = first.compareTo(second);
+		return ascendingComparison ? cmp : -cmp;
+	}
+
+	@Override
+	public boolean invertNormalizedKey() {
+		return !ascendingComparison;
+	}
+
+	@Override
+	public boolean supportsSerializationWithKeyNormalization() {
+		return false;
+	}
+
+	@Override
+	public void writeWithKeyNormalization(LocalDateTime record, DataOutputView target) throws IOException {
+		throw new UnsupportedOperationException();
+	}
+
+	@Override
+	public int extractKeys(Object record, Object[] target, int index) {
+		target[index] = record;
+		return 1;
+	}
+
+	@SuppressWarnings("rawtypes")
+	@Override
+	public TypeComparator[] getFlatComparators() {
+		return comparators;
+	}
+
+	@Override
+	public LocalDateTime readWithKeyDenormalization(LocalDateTime reuse, DataInputView source) throws IOException {
+		throw new UnsupportedOperationException();
+	}
+
+	@Override
+	public int compareSerialized(DataInputView firstSource, DataInputView secondSource) throws IOException {
+		int cmp = dateComparator.compareSerialized(firstSource, secondSource);
+		if (cmp == 0) {
+			cmp = timeComparator.compareSerialized(firstSource, secondSource);
+		}
+		return cmp;
```

Comment:
should we consider ascendingComparison?

Reply:
`this.dateComparator = new LocalDateComparator(ascending);`
The sub comparator has considered the `ascendingComparison`.